### DISCUSSION TITLE
Web下PAGComposition循环播放多个不同时长的PAG文件的疑问

### DISCUSSION THREAD

**Discussion by funfungo** [Tencent/libpag#2008] [Q&A] [answered]
根据文档使用PAGComposition来渲染多个PAG file时，如果两个动画的播放时长不一致，播放时长短的那个文件会在播放时长长的那个文件播放完成后才开始第二个循环
需要的效果应该是每个动画都应该在播放完成后自动开始下一个循环， 请问是否有api可以进行设置呢

https://github.com/Tencent/libpag/assets/2912182/b8a36a26-9bbe-4a68-b708-2c517bad396e

**Comment by kevingpqi123** [comment]
使用下面的接口可以设置单个图层或文件相对于主时间轴的起始时间：
https://pag.art/apis/web/classes/pag_layer.PAGLayer.html#setStartTime

**Reply by funfungo** [reply]
设置起始时间达不到我想要的效果，我想要的是两个动画各自以自己的时间轴进行循环播放，目前的demo是一个动画播完以后要等另一个动画播完才会开始主时间轴的下一个循环。

**Reply by kevingpqi123** [reply, marked as answer]
两种解决方法：
（1）使用两个 PAGView，分别播放；
（2）通过 PAGComposition 组合模式添加两个 pag 文件，设置 PAGFile 的伸缩模式为 repeat，然后设置 PAGFile 的 duration 到一定长度，播放过程中就是循环播放

**Reply by funfungo** [reply]
最后是用duration来设置让两个视频播放时长一致

**Comment by houyinyu** [comment]
我这边是安卓的，之前也遇到相同的问题，具体细节不说了，这边已经和设计说了切两个相同播放时长的动画文件；然后demo里面有一个列表播放，他设置的是pagFile.setDuration，多个动画文件设置相同的duration且足够大，这样一起播放的时候不会出现等待的情况，不知道web端是否有相同的设置或者满足你的需求？
